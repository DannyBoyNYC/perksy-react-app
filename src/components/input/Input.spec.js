const React = require('react');

const Input = require('./index').default;

describe('<Input.* />', () => {
  let component;
  const requiredProps = {
    id: 'foo',
    label: 'Foo label'
  };

  describe('<Input.Checkbox />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Checkbox {...requiredProps} isChecked isDisabled isRequired />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a checked input if isChecked is passed', () => {
      expect(component.find('input')).toBeChecked();
    });

    it('should render a disabled input if isDisabled is passed', () => {
      expect(component.find('input')).toBeDisabled();
    });

    it('should render a required input if isRequired is passed', () => {
      expect(component.find('input')).toHaveProp('required', true);
    });
  });

  describe('<Input.Email />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Email {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.File />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.File {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Hidden />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Hidden {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Number />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Number {...requiredProps} max={80} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Password />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Password {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Radio />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Radio {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Switch />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Switch {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Tel />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(<Input.Tel {...requiredProps} />, 'Input');
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Text />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Text {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Input.Textarea />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Input.Textarea {...requiredProps} />,
        'Input'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
