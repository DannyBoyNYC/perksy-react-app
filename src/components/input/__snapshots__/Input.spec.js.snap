// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input.* /> <Input.Checkbox /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-checkbox-0-1-4 required"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      defaultChecked={true}
      disabled={true}
      id="foo"
      name="foo"
      required={true}
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="checkbox"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Email /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-email-0-1-5"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="email"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.File /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-file-0-1-6"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="file"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Hidden /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-hidden-0-1-7"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="hidden"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Number /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-number-0-1-8"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      max={80}
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="number"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Password /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-password-0-1-9"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="password"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Radio /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-radio-0-1-10"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="radio"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Switch /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-switch-0-1-11"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="checkbox"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Tel /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-tel-0-1-12"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="tel"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Text /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-text-0-1-13"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <input
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="text"
    />
  </label>
</span>
`;

exports[`<Input.* /> <Input.Textarea /> should render correctly with all required props 1`] = `
<span
  className="Input-root-0-1-1 Input-textarea-0-1-14"
>
  <label
    htmlFor="foo"
  >
    <span
      className="Input-label-0-1-3"
    >
      Foo label
    </span>
    <textarea
      className="Input-input-0-1-2"
      id="foo"
      name="foo"
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      type="textarea"
    />
  </label>
</span>
`;
